### DISCUSSION TITLE
Support for HTML, JSX, TSX

### DISCUSSION THREAD

**Discussion by itzcodex24** [LazyVim/LazyVim#2020] [Q&A] [answered]
Hi there, I currently have some issues with auto indent and auto space. For example, when creating a `<div>` by typing it out and then pressing enter, the cursor does not jump between the two tags. Is there a way to fix this functionality? 

Please see the video below for more details into what I mean

https://github.com/LazyVim/LazyVim/assets/70026038/7b4349dc-a348-488b-b04d-c639d298ca54

**Comment by dpetka2001** [comment, marked as answer]
I fixed this by installing plugin `nvim-autopairs` (not sure if there's another way as well, would be glad to hear alternatives).
Put the following in `~/.config/nvim/lua/plugins/whatever.lua` (change `whatever` to the filename you prefer)
```lua
return {
  {
    "hrsh7th/nvim-cmp",
    dependencies = {
      "windwp/nvim-autopairs",
      opts = {},
    },
    opts = function()
      local cmp = require("cmp")
      local cmp_autopairs = require("nvim-autopairs.completion.cmp")
      cmp.event:on("confirm_done", cmp_autopairs.on_confirm_done())
    end,
  },
}
```
Test it to see if it does what you want.

**Reply by itzcodex24** [reply]
This is perfect! Thank you. I remember using the same plugin with my previous configuration, but completely forgot about its existence. Thanks for the reminder :)

**Reply by skoch13** [reply]
Thank you for providing a solution. Should one disable `mini.pairs` to make it work?

**Reply by dpetka2001** [reply]
Not necessarily. I have them both enabled, since I haven't noticed any conflict in my workflow. But you're free to do whatever meets your needs.

**Reply by skoch13** [reply]
Thank you. I'm free :D

**Reply by AnthonyPoschen** [reply]
this just worked perfectly for me, thank you.
for anyone viewing this.
it made
```html
<div>|</div>
```
when pressing enter become the desired
```html
<div>
  |
</div>
```

**Reply by moliyu** [reply]
> I fixed this by installing plugin `nvim-autopairs` (not sure if there's another way as well, would be glad to hear alternatives). Put the following in `~/.config/nvim/lua/plugins/whatever.lua` (change `whatever` to the filename you prefer)
> 
> ```lua
> return {
>   {
>     "hrsh7th/nvim-cmp",
>     dependencies = {
>       "windwp/nvim-autopairs",
>       opts = {},
>     },
>     opts = function()
>       local cmp = require("cmp")
>       local cmp_autopairs = require("nvim-autopairs.completion.cmp")
>       cmp.event:on("confirm_done", cmp_autopairs.on_confirm_done())
>     end,
>   },
> }
> ```
> 
> Test it to see if it does what you want.

how does this work in blink.cmp

**Comment by jjaimealeman** [comment]
I did a fresh install just yesterday of LazyVim. I didn't have to install any plugin to get the functionality you seek. _(or maybe i just couldn't see the video clearly enough?)_

Have you ever used Emmet?
https://docs.emmet.io/cheat-sheet/

Just type out `div` and hit Enter, it creates `<div>|</div>` with your cursor right in the middle.

`nav.ul>li*3`
![image](https://github.com/LazyVim/LazyVim/assets/1428292/3b66c9ed-f2bd-4e8b-82a7-eeb85040eafe)

**Reply by itzcodex24** [reply]
I don't see Emmet as a preinstall into LazyVim, so I would assume that it requires some additional functionality to get it working.

**Reply by jjaimealeman** [reply]
I don't see it installed either, nor did I install it myself.

I'm working with a super fresh installation (less than 24 hours). All I installed were Catppuccin scheme, Copilot, and set Neotree to the right.

Give it a shot amigo `nav.ul>li*3`, it might work out for ya?! 🤔

**Reply by itzcodex24** [reply]
Nope doesn't work, but after adding the `emmet_ls` server to the configuration for lspconfig, it seems to work. 

**NOTE:** The autopairs plugin is still required for the desired functionality outlined in the original post of this discussion

**Reply by dpetka2001** [reply]
If you want to work with `emmet`, you might prefer installing [emmet-language-server](https://github.com/olrtg/emmet-language-server) instead of `emmet-ls`. I've read other ppl on Reddit prefer it, because it implements some features that `emmet-ls` doesn't and it seems to be more maintained for the moment at least.

**Reply by itzcodex24** [reply]
Alright thank you for the idea, will try it out
